Refactoring: introduce MasterEnvironmentRunnableContext to be passed to MasterEnvironmentRunnable

wyzhang opened the pull request.

chtyim commented on cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java:
```diff
@@ -196,6 +197,12 @@ public MasterEnvironmentRunnable createRunnable(MasterEnvironmentContext context
   }
 
+  @Override
+  public MasterEnvironmentRunnable createRunnable(MasterEnvironmentRunnableContext context,
```
Why there are two methods for creating the Runnable? There should only be one.

wyzhang replied:
Existing createRunnable(MasterEnvironmentContext context.....) in MasterEnv is part of SPI. I suppose we cannot remove the existing one.  Marked it as deprecated 

  @Deprecated
  MasterEnvironmentRunnable createRunnable(MasterEnvironmentContext context,
                                           Class<? extends MasterEnvironmentRunnable> runnableClass) throws Exception;

chtyim replied:
We can just remove it. The SPI is very new and we should try to keep it clean for now

wyzhang replied:
Done. Removed

wyzhang committed:
CDAP-17981: mostly refactoring to introduce MasterEnvironmentRunnableContext
which gets passed to MasterEnvironmentRunnable instead of current
MasterEnvironmentRunnable
